cli/pcluster/config/validators.py

```diff
@@ -745,6 +745,14 @@ def scheduler_validator(param_key, param_value, pcluster_config):
     if pcluster_config.get_section("cluster").get_param_value("base_os") not in supported_os:
         errors.append("'{0}' scheduler supports the following Operating Systems: {1}".format(param_value, supported_os))
 
+    deprecated_in_30 = ["sge", "torque"]
+    if param_value in deprecated_in_30:
```

Comment:
An alternative is to use a new attribute to manage the "deprecated" values too. 
e.g.
`"deprecation": {"values": ["sge", "torque"], "version": "3.0.0", "message": "sometext" }`

Then in the validate() function we can have:
```python
from packaging import version
...
def validate(self):

    deprecation = self.definition.get("deprecation", {})
    if deprecation and self.value in deprecation.get("values", []):

        deprecation_version = deprecation.get("version", None)
        if deprecation_version and version.parse(utils.get_installed_version()) < version.parse(deprecation_version):
             message = deprecation.get("message ", None)
             self.pcluster_config.warn(message or "ERROR: Starting in version {0} the value '{1}' for the parameter '{2}' will no longer be supported".format(deprecation_version, self.value, self.key)) 
       else:
           sys.exit("ERROR: Starting from version {0} the value '{1}' is no longer supported for the parameter '{2}".format(deprecation_version, self.value, self.key))
```

The code is a draft and it could be refactored in a specific function `check_deprecation` called by the validate one, but with an approach like this we'll have this information confinated in the `mappings.py` file and we don't need to change the code once we release the 3.0.0.

This kind of code could be helpful if in future we need to deprecate other values.

Library to use: https://packaging.pypa.io/en/latest/version/#packaging.version.parse

Reply:
an alternative is to define a new validator for the deprecation stuff.

Reply:
Per a discussion between Enrico and I, we are going to table this idea for now.